### PR TITLE
feat: support metapipeline

### PR DESCRIPTION
**Pull request by abayer**
Derived from https://github.com/abayer/test-infra/commit/891d910be47cb8f4942a3d575a72276c89d463df, which is tested in https://github.com/jenkins-x/jx/pull/4610

/assign @wbrefvem

### CODE REVIEW

File: cmd/pipeline/controller.go [hunk 1]
```diff
@@ -122,59 +157,62 @@ func (c *controller) hasSynced() bool {
 	return true // Everyone is synced
 }
 
-func newController(opts controllerOptions) (*controller, error) {
-	if err := prowjobscheme.AddToScheme(scheme.Scheme); err != nil {
-		return nil, err
-	}
-
+func newController(kc kubernetes.Interface, pjc prowjobset.Interface, pji prowjobinfov1.ProwJobInformer, pipelineConfigs map[string]pipelineConfig,
+	totURL string, prowConfig config.Getter, rl workqueue.RateLimitingInterface, logger *logrus.Entry) *controller {
```
Comment (wbrefvem) [inline cmd/pipeline/controller.go]: What's the reasoning behind not returning an error here?

Comment (abayer) [reply cmd/pipeline/controller.go]: Beats me! I literally just copypasted it from https://github.com/ccojocar/test-infra/blob/combined-pipelines/prow/cmd/pipeline/controller.go#L163 =)

File: cmd/pipeline/controller.go [hunk 2]
```diff
@@ -122,59 +157,62 @@ func (c *controller) hasSynced() bool {
...
 	// Log to events
+	prowjobscheme.AddToScheme(scheme.Scheme)
```
Comment (wbrefvem) [inline cmd/pipeline/controller.go]: Would like to see `newController` return an error if applicable and preserve the error handling for this call to `AddScheme`.

**Review by wbrefvem — CHANGES_REQUESTED** [review]

File: cmd/pipeline/controller.go (hunk 2 above)
Comment (abayer) [reply cmd/pipeline/controller.go]: Done.

**Review by wbrefvem — APPROVED** [review]

**Pull request merged**